jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java

```diff
@@ -29,4 +33,83 @@ void buildCommonAttributes() {
     assertEquals("test_app_name", attributesMap.get(APP_NAME));
     assertNotNull(attributesMap.get(HOSTNAME));
   }
+
+  @Test
+  void testBuildCommonAttributesWithSpecificHostname() {
+    DaemonConfig config = mock(DaemonConfig.class);
+    when(config.getHostname()).thenReturn("test_hostname");
+    when(config.getMonitoredAppName()).thenReturn("test_app_name");
+
+    Attributes attributes = SetupUtils.buildCommonAttributes(config);
+
+    assertNotNull(attributes);
+    assertEquals("JFR", attributes.asMap().get(INSTRUMENTATION_NAME));
+    assertEquals("JFR-Uploader", attributes.asMap().get(INSTRUMENTATION_PROVIDER));
+    assertEquals("JFR-Uploader", attributes.asMap().get(COLLECTOR_NAME));
+    assertEquals("test_app_name", attributes.asMap().get(SERVICE_NAME));
+    assertEquals("test_app_name", attributes.asMap().get(APP_NAME));
+    assertEquals("test_hostname", attributes.asMap().get(HOSTNAME));
+    assertEquals("test_app_name", attributes.asMap().get(APP_NAME));
```

Comment:
Looks to be a dup of the asserts above (APP_NAME and SERVICE_NAME checks)

Reply:
updated